source/agora/cli/client/GenTxProcess.d

```diff
@@ -0,0 +1,181 @@
+import agora.utils.Test;
+
+import std.algorithm;
+import std.conv;
+import std.format;
+import std.getopt;
+import std.range;
+import std.stdio;
+
+import core.time;
+
+import vibe.core.core;
+
+public alias APIMaker = API delegate (string address);
+
+/// Option required to generate and send transactions
+private struct GenTxOption
+{
+    /// IP address of node
+    public string host;
+
+    /// Port of node
+    public ushort port;
+
+    /// Interval of sending transactions
+    public Duration interval = 5.seconds;
+
+    /// Number of transactions sent at once (1 to 8)
+    public uint count = 1;
+
+    /// dump output option
+    public bool dump;
+}
+
+/// Parse the command-line arguments of gentx
+public GetoptResult parseGenTxOption (ref GenTxOption op, string[] args)
+{
+    return getopt(
+        args,
+        "ip|i",
+            "IP address of node",
+            &op.host,
+
+        "port|p",
+            "Port of node",
+            &op.port,
+
+        "interval|t",
+            "Interval of sending transactions (default: 5 seconds)",
+            (string value) { op.interval = value.to!int.seconds; },
+
+        "count|c",
+            "Number of transactions sent at once (default: 1)",
+            &op.count,
+
+        "dump|o",
+            "Dump output option",
+            &op.dump
+            );
+}
+
+/// Print help
+public void printGenTxHelp (ref string[] outputs)
+{
+    outputs ~= "usage: agora-client gentx [--dump] [--ip] [--port]";
+    outputs ~= "                          [--interval] [--count]";
+    outputs ~= "";
+    outputs ~= "   gentx      Generate and send a transaction to node";
+    outputs ~= "";
+    outputs ~= "        -o --dump       Dump output option";
+    outputs ~= "        -i --ip         IP address of node";
+    outputs ~= "        -p --port       Port of node";
+    outputs ~= "        -t --interval   Interval of sending transactions";
+    outputs ~= "        -c --count      Number of transactions sent at once";
+    outputs ~= "";
+}
+
+/*******************************************************************************
+
+    Input arguments, generate the transaction and send it to the node
+
+    Params:
+        args = client command line arguments
+
+*******************************************************************************/
+
+public int genTxProcess (string[] args, ref string[] outputs,
+                         APIMaker api_maker)
+{
+    GenTxOption op;
+    GetoptResult res;
+
+    try
+    {
+        res = parseGenTxOption(op, args);
+        if (res.helpWanted)
+        {
+            printGenTxHelp(outputs);
+            return CLIENT_SUCCESS;
+        }
+    }
+    catch (Exception ex)
+    {
+        printGenTxHelp(outputs);
+        return CLIENT_EXCEPTION;
+    }
+
+    if (op.count > 8)
+    {
+        printGenTxHelp(outputs);
+        outputs ~= format("Cannot send %s transactions", op.count);
+        return CLIENT_INVALID_ARGUMENTS;
+    }
```

Comment:
Two things:
1) Why ?
2) When a client makes an error, it's not enough to tell them something is wrong. Try to give them a hint at why, and, in case it's not obvious (it is here), how to fix things. Here: "Cannot send more than 8 transactions (%d requested)".